portal/common/lib/types/index.ts

```diff
@@ -25,50 +25,68 @@ export type Resource = {
     headers: Map<string, string>;
     blob_id: string;
     blob_hash: string;
+    range: Range | null;
 };
 
+export type Range = {
+    start: number | null;
+    end: number | null;
+};
+
+function rangeToHttpHeader(range: Range): string {
```

Comment:
Nit: I would add some sanity checks here regarding the values of start and end. 
e.g. if start and end defined, start>=0 end >=start

Reply:
uhm, these are values that are taken from chain. The checks are enforced there. Do we need them here too? 🤔

Reply:
While that is true, I would add the checks because common will be an sdk, and in the future one could use this function in a different context that bypasses those rules.

Reply:
updated in b72764b0629b4a21dd300734a0afc916cccaf5ef